objects/src/transaction/proven_tx.rs

```diff
@@ -0,0 +1,111 @@
+use super::{AccountId, ConsumedNoteInfo, CreatedNoteInfo, Digest, Felt, Hasher, Vec};
+use miden_core::{ProgramInfo, StackInputs, StackOutputs};
+use miden_verifier::{verify, ExecutionProof, VerificationError};
+
+/// Resultant object of executing and proving a transaction. It contains the minimal
+/// amount of data needed to verify the transaction was executed correctly.
+/// Contains:
+/// - account_id: the account that the transaction was executed against.
+/// - initial_account_hash: the hash of the account before the transaction was executed.
+/// - final_account_hash: the hash of the account after the transaction was executed.
+/// - consumed_note_info: a list of consumed notes.
+/// - created_note_info: a list of created notes.
+/// - tx_script_root: the script root of the transaction.
+/// - block_ref: the block hash of the last known block at the time the transaction was executed.
+/// - proof: the proof of the transaction.
+pub struct ProvenTransaction {
+    account_id: AccountId,
+    initial_account_hash: Digest,
+    final_account_hash: Digest,
+    consumed_note_info: Vec<ConsumedNoteInfo>,
+    created_note_info: Vec<CreatedNoteInfo>,
+    tx_script_root: Digest,
+    block_ref: Digest,
+    proof: ExecutionProof,
+}
+
+impl ProvenTransaction {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        account_id: AccountId,
+        initial_account_hash: Digest,
+        final_account_hash: Digest,
+        consumed_note_info: Vec<ConsumedNoteInfo>,
+        created_note_info: Vec<CreatedNoteInfo>,
+        tx_script_root: Digest,
+        block_ref: Digest,
+        proof: ExecutionProof,
+    ) -> Self {
+        Self {
+            account_id,
+            initial_account_hash,
+            final_account_hash,
+            consumed_note_info,
+            created_note_info,
+            tx_script_root,
+            block_ref,
+            proof,
+        }
+    }
+
+    /// Verify the transaction using the provided data and proof.
+    pub fn verify(&self) -> Result<u32, VerificationError> {
+        // create verification dependencies
+        let stack_inputs = self.stack_inputs();
+        let stack_outputs = self.stack_outputs();
+        let program_info = ProgramInfo::new(self.tx_script_root(), Default::default());
```

Comment:
Instead of building `program_info` here, we should have a separate accessor - something like:
```Rust
pub fn tx_program(&self) -> ProgramInfo {
    todo!()
}
```
Eventually, this will use note script roots, tx kernel, and tx_script to build a single program for the transaction.